llvm/utils/lit/lit/TestRunner.py

```diff
@@ -1005,6 +1005,44 @@ def _executeShCmd(cmd, shenv, results, timeoutHelper):
     return exitCode
 
 
+def reFindLast(r, s):
+    matches = r.findall(s)
```

Comment:
Can we do this without a regex in the common case, just with .rfind("\33["). And only if there is actually a match in the line check that it has the expected pattern (either with or without a regex).